R-Interface/R/symlinkTools.R

```diff
@@ -0,0 +1,302 @@
+    return(tgtToSrc)
+  }
+
+  #This one returns the path from the root (overlap) to where target is.
+  sourceToTarget <- function(target)
+  {
+    targetSplit  <- splitPath(target)
+    srcToTgt     <- pastePath(targetSplit[seq(overlap$len + 1, length(targetSplit))])
+
+    return(srcToTgt)
+  }
+
+  return(list(targetToSource=targetToSource, sourceToTarget=sourceToTarget))
+}
+
+#Use overlapfunctions as returned by determineOverlap to generate a function to turn target-path from absolute to relative
+getRelativityFunction <- function(modulesRoot, renvCache)
+{
+  #I wanted this code to be more general but then it is too complicated to debug. And has a bug. And renv-cache can be assumed to be right next to Modules anyway...
+  # So instead of doing:
+  #   modToRenvF <- determineOverlap(modulesRoot, renvCache)
+  #   modToRenvS <-  modToRenvF$targetToSource(renvCache, TRUE)
+  # We can assume:
+  modToRenvS <- pastePath(c("..","..","renv-cache"))
+  #print(paste0("modToRenvS: ", modToRenvS))
+  
+  return(
+    function(linkLocation, targetPath)
+    {
+      linkLocation <-              (path.expand(linkLocation)) #Do not normalize it because Windows then follows this path...
+      targetPath   <- normalizePath(path.expand(targetPath))
+      #linkToMod    <- determineOverlap(linkLocation, modulesRoot)$targetToSource
+      pathToRenv   <- determineOverlap(targetPath,   renvCache)  $sourceToTarget
+
+      #linkToModS   <- linkToMod(linkLocation, FALSE)
+      linkToRenvS  <- modToRenvS #pastePath(c(linkToModS, modToRenvS))
+      pathToRenvS  <- pathToRenv(targetPath)
+
+      newTarget    <- paste0(linkToRenvS, .Platform$file.sep, pathToRenvS)
+
+      #print(paste0("for link '", linkLocation, "' and target '",targetPath, "'"))
+      #print(paste0("- linkToModS '",linkToModS, " modToRenvS: '", modToRenvS, "' pathToRenvS: '", pathToRenvS, "'\n results in newTarget: '", newTarget, "'"))
+      
+      return(newTarget)
+    }
+  )
+}
+
+ # copy paste from https://stat.ethz.ch/R-manual/R-patched/library/base/html/Sys.readlink.html
+is.symlink  <- function(paths) isTRUE(nzchar(Sys.readlink(paths), keepNA=TRUE))
+
+# See https://stackoverflow.com/questions/2311105/test-in-powershell-code-if-a-folder-is-a-junction-point because there is no R equivalent...
+# these functions are also really quite slow... But ok, it is only necessary to run it during building.
+is.junction <- function(paths)
+{
+  #Also this function isn't very safe, make sure that you only ask it about existing paths or you get garbage
+  as.logical(
+    sapply(
+      paths, 
+      function(path) 
+        as.logical(as.integer(
+          system2(command="powershell", args=paste0('-command "if ((Get-Item -Path ', path,' -Force).LinkType -eq \\"Junction\\") { 1 } else { 0 }'), stdout=TRUE)
+        )) 
+    )
+  )
+}
+
+get.junction.follow <- function(paths) 
+{
+  sapply(
+    paths, 
+    function(path) normalizePath(path) #Works fine but only if the target exists
+)
```

Comment:
`normalizePaths` is vectorized
```suggestion
  existing <- file.exists(paths)
  if (!all(existing))
    warning("Some paths passed to get.junction.follow did not exist!")
  paths[existing] <- normalizePaths(paths[existing])
)
```

Reply:
Aah ok then I can just remove the whole function, because where it is used it is certain it exists. And if not it should totally break.